### PR TITLE
TF-5: add functionality for working with roles

### PR DESCRIPTION
**Pull request by alex122dev**

### CODE REVIEW

File: server/db/seeds/AdminRoleSeeder.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        }
+
+        const adminExists = await rolesRepository.findOneBy({ value: adminData.value })
+
+        if (!adminExists) {
+            const admin = rolesRepository.create(adminData)
+            await rolesRepository.save(admin)
+        }
+    }
+}
```
Comment (mbchars) [inline server/db/seeds/AdminRoleSeeder.ts]: EOF

Comment (mbchars) [reply server/db/seeds/AdminRoleSeeder.ts]: Add /db to eslint

File: server/src/domains/roles/roles.guard.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { Observable } from 'rxjs';
+import { AuthService } from '../auth/auth.service';
+import { ROLES_KEY } from './roles.decorator';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector, private authService: AuthService) {}
```
Comment (mbchars) [inline server/src/domains/roles/roles.guard.ts]: Do not use token

File: server/src/domains/users/dto/update-users-roles.dto.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export class UpdateUsersRolesDto {
```
Comment (mbchars) [inline server/src/domains/users/dto/update-users-roles.dto.ts]: Propose rename -> UpdateUserRoleDto

File: server/src/domains/users/dto/user.dto.ts [hunk 4]
```diff
@@ -6,12 +7,14 @@ export class UserDto {
   firstName: string;
   lastName: string;
   phone: string;
+  roles: Role[];
```
Comment (mbchars) [inline server/src/domains/users/dto/user.dto.ts]: Propose to change to RoleDto

File: server/db/seeds/MainSeeder.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+import { runSeeder, Seeder, SeederFactoryManager } from "typeorm-extension";
+import { AdminRoleSeeder } from "./AdminRoleSeeder";
+import { UserRoleSeeder } from "./UserRoleSeeder";
+
+export class MainSeeder implements Seeder {
+    async run(dataSource: DataSource, factoryManager: SeederFactoryManager): Promise<void> {
+        await runSeeder(dataSource, AdminRoleSeeder)
+        await runSeeder(dataSource, UserRoleSeeder)
+    }
+}
```
Comment (mbchars) [inline server/db/seeds/MainSeeder.ts]: EOF

File: server/src/domains/roles/roles.guard.ts [hunk 6]
```diff
@@ -27,30 +36,7 @@ export class RolesGuard implements CanActivate {
...
-      const userPayload = this.authService.getUserPayloadFromToken(accessToken);
-
-      if (!userPayload) {
-        throw new UnauthorizedException();
-      }
-
-      req.user = userPayload;
-
-      const isAccess = userPayload.roles.some((role) =>
+      const isAccess = user.roles.some((role) =>
```
Comment (mbchars) [inline server/src/domains/roles/roles.guard.ts]: What if required roles are:

- ADMIN
- ACCOUNTANT


But user is only ADMIN.

Example: a page with financial data, where only the admin can get and only the accountant can do something with the money on this page.

**Review by nikitabls — APPROVED** [review]

**Review by mbchars — APPROVED** [review]

**Pull request merged**